compiler/src/iree/compiler/InputConversion/MHLO/MHLOToMHLOPreprocessing.cpp

```diff
@@ -778,6 +778,69 @@ struct FuseWidenOperands : public OpRewritePattern<Op> {
   }
 };
 
+struct DotToMul : public OpRewritePattern<mhlo::DotOp> {
+  using OpRewritePattern<mhlo::DotOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(mhlo::DotOp op,
+                                PatternRewriter &rewriter) const override {
+    auto lhs = op.lhs();
+    auto rhs = op.rhs();
+    auto lhsTy = lhs.getType().cast<RankedTensorType>();
+    auto rhsTy = rhs.getType().cast<RankedTensorType>();
+    auto resultTy = op.getType().cast<RankedTensorType>();
+
+    if (lhsTy.getDimSize(1) != 1) return failure();
+
+    Value batchSize = rewriter.create<mhlo::GetDimensionSizeOp>(
```

Comment:
Could you add a high level description here?

Reply:
Done.